platform/darwin/src/MGLShapeSource.mm

```diff
@@ -99,8 +100,15 @@ - (void)commonInit
         source->setGeoJSON(geojson);
         _shape = MGLShapeFromGeoJSON(geojson);
     } else {
-        const auto geojson = mbgl::GeoJSON{self.shape.geometryObject};
-        source->setGeoJSON(geojson);
+        if ([self.shape isMemberOfClass:[MGLShapeCollectionFeature class]]) {
```

Comment:
Using `-isKindOfClass:` will also avoid latent bugs if we ever end up needing to subclass MGLShapeCollectionFeature ourselves.